lib/knock/authenticable.rb

```diff
@@ -9,7 +9,7 @@ module Knock
   # the token payload.
   module Authenticable
     def authenticate_for(entity_class)
-      getter_name = "current_#{entity_class.to_s.parameterize.underscore}"
+      getter_name = "current_#{entity_class.to_s.underscore.parameterize}"
```

Comment:
Please, can you add unity tests to some possible cases for this?

Reply:
Added tests and replaced `parameterize` to `gsub('::', '')`